### PR TITLE
layout of the login page

### PR DESCRIPTION
**Pull request by Rey-2903**

### CODE REVIEW

File: apps/schools/domains/common/components/Anchor/index.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { Anchor as DefaultAnchor } from 'antd'
+import React from 'react'
+
+interface CustomAnchorProps {
+    type?: 'anchorDefault',
+    onClick?: () => void,
+    text?: string,
+}
```
Comment (levil664) [inline apps/schools/domains/common/components/Anchor/index.tsx]: need ...restProps like these
```
import { ButtonHTMLAttributes } from 'react'

interface CustomButtonProps extends Omit<ButtonHTMLAttributes<HTMLButtonElement>, 'type'> {
    type?: 'schoolDefault' | 'schoolGradient';
    antdType?: ButtonProps['type'];
}

export const Button: React.FC<CustomButtonProps> = (props) => {
    const { type = 'schoolDefault', antdType, ...restProps } = props
```

File: apps/schools/domains/common/components/Anchor/index.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        <div>
+            <a>Регистрация</a>
+            <a>Вход</a>
+        </div>
+    )
+}
+
+export const Anchor: React.FC<CustomAnchorProps> = (props) => {
+    const { type = 'anchorDefault', onClick } = props
+
```
Comment (levil664) [inline apps/schools/domains/common/components/Anchor/index.tsx]: are needed ...otherProps

File: apps/schools/domains/common/components/Anchor/index.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+const anchorItems = () => {
+    return (
+        <div>
+            <a>Регистрация</a>
+            <a>Вход</a>
+        </div>
+    )
+}
```
Comment (levil664) [inline apps/schools/domains/common/components/Anchor/index.tsx]: must be in constants folder

File: apps/schools/domains/common/components/Button/index.tsx [hunk 4]
```diff
@@ -4,38 +4,31 @@ import defaultStyles from './styles/default.module.scss'
...
-}
-
-const textButton = () => {
-    return (
-        <div>
-            Hello world!
-        </div>
-    )
+    'schoolDefault': defaultStyles,
+    'schoolGradient': gradientStyles,
```
Comment (levil664) [inline apps/schools/domains/common/components/Button/index.tsx]: we use gradient styles?

File: apps/schools/domains/common/components/Anchor/index.tsx (hunk 1 above)
Comment (levil664) [reply apps/schools/domains/common/components/Anchor/index.tsx]: or these 
```
import { ButtonProps } from 'antd'

interface CustomButtonProps extends Omit<ButtonProps, 'type'> {
    type?: 'schoolDefault' | 'schoolGradient',
    antdType?: ButtonProps['type'],
    onClick?: () => void,
    text?: string,
}


export const Button: React.FC<CustomButtonProps> = (props) => {
    const { type = 'schoolDefault', antdType = 'primary', onClick, ...restProps } = props
```

File: apps/schools/domains/common/components/CentralizingContainer/index.tsx [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import React, { PropsWithChildren } from 'react'
+import { Row } from 'antd'
+import styles from './styles/styles.module.scss'
+
+export const CentralizingContainer: React.FC<any> = (props: PropsWithChildren<any>) => {
```
Comment (levil664) [inline apps/schools/domains/common/components/CentralizingContainer/index.tsx]: there shouldn't be any

Comment (levil664) [reply apps/schools/domains/common/components/CentralizingContainer/index.tsx]: somehow, but we should avoid any

File: apps/schools/domains/common/components/CentralizingContainer/index.tsx
Comment (levil664) [inline apps/schools/domains/common/components/CentralizingContainer/index.tsx:1]: why CentralizingContainer in common domain? And if it should be here, make it normally reused with ...restProps.

File: apps/schools/domains/common/components/CentralizingContainer/index.tsx [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+    const { children } = props
+    return (
+        <div className={styles.fullHeight}>
+            <Row justify='center' align="middle" className={styles.fullHeight}>
```
Comment (levil664) [inline apps/schools/domains/common/components/CentralizingContainer/index.tsx]: fullHeight? And if we have to add something else?

File: apps/schools/domains/common/components/CentralizingContainer/styles/styles.module.scss [hunk 7]
```diff
@@ -0,0 +1,3 @@
+.fullHeight{
```
Comment (levil664) [inline apps/schools/domains/common/components/CentralizingContainer/styles/styles.module.scss]: ^^

File: apps/schools/domains/common/components/CountDownTimer/index.tsx
Comment (levil664) [inline apps/schools/domains/common/components/CountDownTimer/index.tsx:1]: why CountDownTimer in common domain? And if it should be here, make it normally reused with ...restProps.

File: apps/schools/domains/common/components/CountDownTimer/index.tsx [hunk 8]
```diff
@@ -0,0 +1,115 @@
+import get from 'lodash/get'
+import React, { useEffect, useState } from 'react'
+
+import { extractRootDomain } from '../../utils/url.utils'
+
+import { timer } from '../../utils/timer'
```
Comment (levil664) [inline apps/schools/domains/common/components/CountDownTimer/index.tsx]: why are there spaces between imports?

File: apps/schools/domains/common/components/CountDownTimer/index.tsx [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+
+const DEFAULT_TIMEOUT = 60 // s
```
Comment (levil664) [inline apps/schools/domains/common/components/CountDownTimer/index.tsx]: must be in constants

File: apps/schools/domains/common/components/CountDownTimer/index.tsx [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+
+const setCountDownDate = (date: any, countDownId: any) => {
+    const domain = extractRootDomain(window.location.href)
+
+    document.cookie = `COUNTDOWN_${countDownId}=${date}; path=/; domain=${domain}`
+}
```
Comment (levil664) [inline apps/schools/domains/common/components/CountDownTimer/index.tsx]: must be in utils

File: apps/schools/domains/common/components/CountDownTimer/index.tsx [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+        const now = new Date()
+        const countDownDate = new Date(countDownFromCookie)
+
+        if (countDownDate < now) {
+            return 0
+        } else {
+            return new Date(Number(countDownDate) - Number(now)).getSeconds()
+        }
+    }
+}
```
Comment (levil664) [inline apps/schools/domains/common/components/CountDownTimer/index.tsx]: ^^

File: apps/schools/domains/common/components/CountDownTimer/index.tsx [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+) => JSX.Element
+
+interface ICountDownTimer {
+    // TODO(Dimitreee): remove any
+    action: (...args: any[]) => Promise<any>
+    id: string
+    timeout?: number
+    children: CountDownChildrenType
+    autostart?: boolean
+}
```
Comment (levil664) [inline apps/schools/domains/common/components/CountDownTimer/index.tsx]: we have an Interfaces folder

File: apps/schools/domains/common/components/CountDownTimer/index.tsx [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+            },
+            onUpdate: (currentDuration) => {
+                setCountDown(currentDuration)
+            },
+            onFinish: () => {
+                setCountDown(0)
+                setCountDownDate(0, id)
+            },
+        })
+    }, [])
```
Comment (levil664) [inline apps/schools/domains/common/components/CountDownTimer/index.tsx]: looks like must be in utils or tasks folder

File: apps/schools/domains/common/components/CountDownTimer/index.tsx [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+        return action()
+            .then((res) => {
+                setLoading(false)
+                return res
+            })
+            .catch((e) => {
+                setLoading(false)
+                throw e
+            })
+    }
```
Comment (levil664) [inline apps/schools/domains/common/components/CountDownTimer/index.tsx:100]: ^^

File: apps/schools/domains/common/components/Input/index.tsx [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+interface Dictionary {
+    [key: string]: any;
+}
+
+const inputStyleDictionary : Dictionary = {
+    'inputDefault': defaultStyles,
+    'inputPassword': passwordStyle,
+}
```
Comment (levil664) [inline apps/schools/domains/common/components/Input/index.tsx]: we have an Interfaces folder

File: apps/schools/domains/common/components/Input/index.tsx [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+
+const INPUT_PHONE_STYLE: React.CSSProperties = { width: "100%", height: 48, borderRadius: 12, borderColor: '#D9D9D9' }
+const BUTTON_INPUT_PHONE_STYLE: React.CSSProperties = { margin: 5, backgroundColor: '#E6E8F1', border: 0, borderRadius: 8 }
```
Comment (levil664) [inline apps/schools/domains/common/components/Input/index.tsx]: styles.module.scss?

File: apps/schools/domains/common/components/Input/index.tsx [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+    } else if (type === 'inputPhone') {
+        return (
+            <div className={defaultStyles.inputContainer}>
+                <label>{label}</label>
+                <PhoneInput
+                    country={'ru'}
+                    placeholder={placeholder}
+                    onChange={onChange}
+                    buttonStyle={BUTTON_INPUT_PHONE_STYLE}
+                    inputStyle={INPUT_PHONE_STYLE}
```
Comment (levil664) [inline apps/schools/domains/common/components/Input/index.tsx:53]: ^^

File: apps/schools/domains/common/components/Link/index.tsx [hunk 18]
```diff
@@ -0,0 +1,30 @@
+import React from 'react'
+import defaultStyles from './styles/default.module.scss'
+import underliningStyles from './styles/underlining.module.scss'
+import { typeLink } from '../../constants/Link'
+
+interface CustomButtonProps {
+    type?: 'linkDefault' | 'linkUnderlining',
+    onClick?: () => void,
+    text?: string,
+}
```
Comment (levil664) [inline apps/schools/domains/common/components/Link/index.tsx]: need ...restProps like these
```
import { ButtonHTMLAttributes } from 'react'

interface CustomButtonProps extends Omit<ButtonHTMLAttributes, 'type'> {
type?: 'schoolDefault' | 'schoolGradient';
antdType?: ButtonProps['type'];
}

export const Button: React.FC = (props) => {
const { type = 'schoolDefault', antdType, ...restProps } = props
```

or these
```
import { ButtonProps } from 'antd'

interface CustomButtonProps extends Omit<ButtonProps, 'type'> {
type?: 'schoolDefault' | 'schoolGradient',
antdType?: ButtonProps['type'],
onClick?: () => void,
text?: string,
}

export const Button: React.FC = (props) => {
const { type = 'schoolDefault', antdType = 'primary', onClick, ...restProps } = props
```

File: apps/schools/domains/common/components/Link/index.tsx [hunk 19]
```diff
@@ -0,0 +1,30 @@
...
+
+interface Dictionary {
+    [key: string]: any;
+}
+
+const linkStyleDictionary : Dictionary = {
+    'linkDefault': defaultStyles,
+    'linkUnderlining': underliningStyles,
+}
```
Comment (levil664) [inline apps/schools/domains/common/components/Link/index.tsx]: interfaces folder?

File: apps/schools/domains/common/components/Anchor/index.tsx [hunk 20]
```diff
@@ -0,0 +1,26 @@
+import { Anchor as DefaultAnchor } from 'antd'
+import React from 'react'
+
+interface CustomAnchorProps {
+    type?: 'anchorDefault',
+    onClick?: () => void,
+    text?: string,
+}
+
```
Comment (levil664) [inline apps/schools/domains/common/components/Anchor/index.tsx]: interfaces folder?

File: apps/schools/domains/common/components/Button/index.tsx [hunk 21]
```diff
@@ -4,38 +4,31 @@ import defaultStyles from './styles/default.module.scss'
...
 }
```
Comment (levil664) [inline apps/schools/domains/common/components/Button/index.tsx]: interfaces folder?

File: apps/schools/domains/common/components/CountDownTimer/index.tsx [hunk 22]
```diff
@@ -0,0 +1,115 @@
+import get from 'lodash/get'
+import React, { useEffect, useState } from 'react'
+
+import { extractRootDomain } from '../../utils/url.utils'
+
+import { timer } from '../../utils/timer'
+
+const DEFAULT_TIMEOUT = 60 // s
+
+const setCountDownDate = (date: any, countDownId: any) => {
```
Comment (levil664) [inline apps/schools/domains/common/components/CountDownTimer/index.tsx]: we should not use any

File: apps/schools/domains/common/components/Link/styles/underlining.module.scss [hunk 23]
```diff
@@ -0,0 +1,3 @@
+.link {
+
+}
```
Comment (levil664) [inline apps/schools/domains/common/components/Link/styles/underlining.module.scss]: for what?

File: apps/schools/domains/common/utils/helpers.ts [hunk 24]
```diff
@@ -0,0 +1,87 @@
+import { ParsedUrlQuery } from 'querystring'
+
+
```
Comment (levil664) [inline apps/schools/domains/common/utils/helpers.ts]: why two spaces?

File: apps/schools/domains/common/utils/helpers.ts [hunk 25]
```diff
@@ -0,0 +1,87 @@
...
+import get from 'lodash/get'
+import isArray from 'lodash/isArray'
+import isEmpty from 'lodash/isEmpty'
+import isNumber from 'lodash/isNumber'
+import isObject from 'lodash/isObject'
+import { NextRouter } from 'next/router'
+import qs from 'qs'
+
+const DEFAULT_WIDTH_PRECISION = 2
+const PHONE_FORMAT_REGEXP = /(\d)(\d{3})(\d{3})(\d{2})(\d{2})/
```
Comment (levil664) [inline apps/schools/domains/common/utils/helpers.ts:12]: constants folder?

File: apps/schools/domains/common/utils/helpers.ts [hunk 26]
```diff
@@ -0,0 +1,87 @@
...
+
+interface IUpdateQueryOptions {
+    routerAction: 'replace' | 'push'
+    resetOldParameters: boolean
+}
+interface IUpdateQueryData {
+    newParameters: Record<string, unknown>
+    newRoute: string
+}
+type UpdateQueryType = (router: NextRouter, data: Partial<IUpdateQueryData>, options?: Partial<IUpdateQueryOptions>) => Promise<void>
```
Comment (levil664) [inline apps/schools/domains/common/utils/helpers.ts]: interfaces folder?

File: apps/schools/domains/common/utils/helpers.ts [hunk 27]
```diff
@@ -0,0 +1,87 @@
...
+
+export const updateQuery: UpdateQueryType = async (router, data, options) => {
+    const newParameters: any = get(data, 'newParameters', {})
```
Comment (levil664) [inline apps/schools/domains/common/utils/helpers.ts:69]: we should not use any

File: apps/schools/domains/common/utils/timer.ts [hunk 28]
```diff
@@ -0,0 +1,41 @@
+// refs to: https://www.w3schools.com/howto/howto_js_countdown.asp
+interface ITimerArguments {
+    onUpdate: (progress: number) => void
+    onStart?: (startDate: Date) => void
+    onFinish?: () => void
+    duration: number // s
+    step?: number // ms
+}
```
Comment (levil664) [inline apps/schools/domains/common/utils/timer.ts:8]: interfaces folder?

File: apps/schools/domains/user/components/auth/FormContainer/index.tsx
Comment (levil664) [inline apps/schools/domains/user/components/auth/FormContainer/index.tsx:1]: are we sure we don't need ...restProps in domains except common?

File: apps/schools/domains/user/components/auth/FormContainer/index.tsx [hunk 29]
```diff
@@ -0,0 +1,21 @@
+import styles from '../styles/formStyles.module.scss'
+import { Col, Row } from 'antd'
+import React, { Children, PropsWithChildren } from 'react'
+
+export const FormContainer: React.FC<any> = (props: PropsWithChildren<any>) => {
```
Comment (levil664) [inline apps/schools/domains/user/components/auth/FormContainer/index.tsx]: we must not use any

File: apps/schools/domains/user/components/auth/FormContainer/styles/styles.module.scss
Comment (levil664) [inline apps/schools/domains/user/components/auth/FormContainer/styles/styles.module.scss:1]: empty

File: apps/schools/domains/user/components/auth/HeaderActions/index.tsx [hunk 30]
```diff
@@ -0,0 +1,31 @@
+import { Tabs } from 'antd'
+import Router from 'next/router'
+import React from 'react'
+
+import styles from './styles/styles.module.scss'
+
+interface ITabsActionsProps {
+    currentActiveKey: string
+}
```
Comment (levil664) [inline apps/schools/domains/user/components/auth/HeaderActions/index.tsx]: must be in interface folder

File: apps/schools/domains/user/components/auth/HeaderActions/index.tsx [hunk 31]
```diff
@@ -0,0 +1,31 @@
...
+
+export const TabsAuthAction: React.FC<ITabsActionsProps> = (props) => {
+    const { currentActiveKey } = props
+
+    const registerTab = 'Регистрация'
+    const signInTab = 'Вход'
```
Comment (levil664) [inline apps/schools/domains/user/components/auth/HeaderActions/index.tsx:12]: why won 't we put it in constants?

File: apps/schools/domains/user/components/auth/RegisterForm/RegisterForm.tsx [hunk 32]
```diff
@@ -0,0 +1,170 @@
...
+
+import styles from '../styles/formStyles.module.scss'
+import { ResponsiveCol } from '../../containers/ResponsiveCol'
+import { Input } from '../../../../common/components/Input'
+import { Button } from '../../../../common/components'
+
+
+interface IRegisterFormProps {
+    onFinish: (userId: string) => void
+}
```
Comment (levil664) [inline apps/schools/domains/user/components/auth/RegisterForm/RegisterForm.tsx]: interfaces folder?

File: apps/schools/domains/user/components/auth/RegisterForm/RegisterForm.tsx [hunk 33]
```diff
@@ -0,0 +1,170 @@
...
+
+const BUTTON_FORM_GUTTER: RowProps['gutter'] = [0, 40]
```
Comment (levil664) [inline apps/schools/domains/user/components/auth/RegisterForm/RegisterForm.tsx]: constants folder?

File: apps/schools/domains/user/components/auth/RegisterForm/RegisterForm.tsx [hunk 34]
```diff
@@ -0,0 +1,170 @@
...
+
+const RequiredFlagWrapper: React.FC<PropsWithChildren<any>> = (props) => {
+    return (
+        <div className={styles.requiredField}>
+            {props.children}
+        </div>
+    )
+}
+
+import { useRegisterFormValidators } from './hooks'
```
Comment (levil664) [inline apps/schools/domains/user/components/auth/RegisterForm/RegisterForm.tsx]: why this import here?

File: apps/schools/domains/common/components/CountDownTimer/index.tsx [hunk 35]
```diff
@@ -0,0 +1,115 @@
...
+    autostart?: boolean
+}
+
+export const CountDownTimer: React.FC<ICountDownTimer> = (props) => {
+    const { action, id, timeout = DEFAULT_TIMEOUT, autostart = false } = props
+
+    const [loading, setLoading] = useState(false)
+    const [countdown, setCountDown] = useState(0)
+
+    const startTimer = React.useCallback((duration) => {
```
Comment (levil664) [inline apps/schools/domains/common/components/CountDownTimer/index.tsx]: why is it react.useCallback and not just useCallback

Comment (levil664) [reply apps/schools/domains/common/components/CountDownTimer/index.tsx]: in other places you used useCallback

File: apps/schools/domains/user/components/auth/RegisterForm/hooks.tsx [hunk 36]
```diff
@@ -0,0 +1,64 @@
+import { Rule } from 'rc-field-form/lib/interface'
+import { useMemo } from 'react'
+
+type ValidatorsMap = {
+    [key: string]: Rule[]
+}
```
Comment (levil664) [inline apps/schools/domains/user/components/auth/RegisterForm/hooks.tsx]: interfaces folder?

File: apps/schools/domains/user/components/auth/RegisterForm/hooks.tsx
Comment (levil664) [inline apps/schools/domains/user/components/auth/RegisterForm/hooks.tsx:1]: hooks folder?

File: apps/schools/domains/user/components/auth/ResetForm/ResetForm.tsx [hunk 37]
```diff
@@ -0,0 +1,132 @@
...
+
+import styles from '../styles/formStyles.module.scss'
+import { ResponsiveCol } from '../../containers/ResponsiveCol'
+import { Input } from '../../../../common/components/Input'
+import { Button } from '../../../../common/components'
+
+
+interface IResetFormProps {
+    onFinish: (userId: string) => void
+}
```
Comment (levil664) [inline apps/schools/domains/user/components/auth/ResetForm/ResetForm.tsx]: interfaces folder?

File: apps/schools/domains/user/components/auth/ResetForm/ResetForm.tsx [hunk 38]
```diff
@@ -0,0 +1,132 @@
...
+
+const BUTTON_FORM_GUTTER: RowProps['gutter'] = [0, 40]
```
Comment (levil664) [inline apps/schools/domains/user/components/auth/ResetForm/ResetForm.tsx]: constants folder?

File: apps/schools/domains/user/components/auth/ResetForm/ResetForm.tsx [hunk 39]
```diff
@@ -0,0 +1,132 @@
...
+
+const RequiredFlagWrapper: React.FC<PropsWithChildren<any>> = (props) => {
+    return (
+        <div className={styles.requiredField}>
+            {props.children}
+        </div>
+    )
+}
+
+import { useResetFormValidators } from './hooks'
```
Comment (levil664) [inline apps/schools/domains/user/components/auth/ResetForm/ResetForm.tsx]: why this import here?

File: apps/schools/domains/user/components/auth/ResetForm/hooks.tsx
Comment (levil664) [inline apps/schools/domains/user/components/auth/ResetForm/hooks.tsx:1]: hooks folder?

File: apps/schools/domains/user/components/auth/ResetForm/hooks.tsx [hunk 40]
```diff
@@ -0,0 +1,64 @@
+import { Rule } from 'rc-field-form/lib/interface'
+import { useMemo } from 'react'
+
+type ValidatorsMap = {
+    [key: string]: Rule[]
+}
```
Comment (levil664) [inline apps/schools/domains/user/components/auth/ResetForm/hooks.tsx]: interfaces folder?

File: apps/schools/domains/user/components/auth/SharedForms/InputPhoneForm.tsx [hunk 41]
```diff
@@ -0,0 +1,135 @@
...
+import styles from '../styles/formStyles.module.scss'
+
+import { Button } from '../../../../common/components'
+import { normalizePhone } from '../../../../common/utils/phone'
+
+interface IInputPhoneFormProps {
+    onFinish: () => void,
+    nextUrl: string,
+    title: string,
+}
```
Comment (levil664) [inline apps/schools/domains/user/components/auth/SharedForms/InputPhoneForm.tsx]: interfaces folder?

File: apps/schools/domains/user/components/auth/SharedForms/ValidatePhoneForm.tsx [hunk 42]
```diff
@@ -0,0 +1,191 @@
...
+import { formatPhone } from '../../../../common/utils/helpers'
+import { ResponsiveCol } from '../../containers/ResponsiveCol'
+import { Input } from '../../../../common/components/Input'
+import { CountDownTimer } from '../../../../common/components/CountDownTimer'
+
+interface IValidatePhoneFormProps {
+    onFinish: () => void,
+    onReset: () => void,
+    title: string,
+}
```
Comment (levil664) [inline apps/schools/domains/user/components/auth/SharedForms/ValidatePhoneForm.tsx]: interfaces folder?

File: apps/schools/domains/user/components/auth/SharedForms/ValidatePhoneForm.tsx [hunk 43]
```diff
@@ -0,0 +1,191 @@
...
+
+const SMS_CODE_CLEAR_REGEX = /[^0-9]/g
+const BUTTON_FORM_GUTTER: RowProps['gutter'] = [0, 40]
+const FORM_ITEMS_GUTTER: RowProps['gutter'] = [0, 24]
```
Comment (levil664) [inline apps/schools/domains/user/components/auth/SharedForms/ValidatePhoneForm.tsx]: constants folder?

File: apps/schools/domains/user/components/auth/SharedForms/ValidatePhoneForm.tsx [hunk 44]
```diff
@@ -0,0 +1,191 @@
...
+const INPUT_STYLE: CSSProperties = { maxWidth: '120px' }
```
Comment (levil664) [inline apps/schools/domains/user/components/auth/SharedForms/ValidatePhoneForm.tsx]: styles.module.scss?

File: apps/schools/domains/user/components/auth/SignInForm/styles/styles.module.scss
Comment (levil664) [inline apps/schools/domains/user/components/auth/SignInForm/styles/styles.module.scss:1]: empty

File: apps/schools/domains/user/components/auth/SharedForms/ValidatePhoneForm.tsx [hunk 45]
```diff
@@ -0,0 +1,191 @@
...
+        // if (smsCode.length > SMS_CODE_LENGTH) {
+        //     return setPhoneValidateError(SMSCodeMismatchError)
+        // }
+        try {
+            await confirmPhone()
+            onFinish()
+        } catch (error) {
+            console.error(error)
+        }
+    }, [confirmPhone])
```
Comment (levil664) [inline apps/schools/domains/user/components/auth/SharedForms/ValidatePhoneForm.tsx]: we have tasks folder for async and callback functions.

File: apps/schools/domains/user/components/containers/AuthLayout.tsx [hunk 46]
```diff
@@ -0,0 +1,73 @@
...
+    element: 'ReCaptchaContainer',
+    parameters: {
+        badge: 'inline',
+    },
+}
+const GOOGLE_RECAPTCHA_SCRIPT_PROPS: IGoogleReCaptchaScriptProps = {
+    async: true,
+    defer: true,
+    appendTo: 'body',
+}
```
Comment (levil664) [inline apps/schools/domains/user/components/containers/AuthLayout.tsx]: interfaces and constants folder?

File: apps/schools/domains/user/components/containers/AuthLayout.tsx [hunk 47]
```diff
@@ -0,0 +1,73 @@
...
+
+// #TODO: take this variables from env
+const {
+    publicRuntimeConfig: { HelpRequisites: { support_email: SUPPORT_EMAIL = null, support_phone: SUPPORT_PHONE = null } },
+} /* = getConfig() */ = { publicRuntimeConfig: { HelpRequisites: { support_email: 'openschools@education', support_phone: '+79999999999' } } }
```
Comment (levil664) [inline apps/schools/domains/user/components/containers/AuthLayout.tsx]: constants folder?

File: apps/schools/pages/_app.tsx [hunk 48]
```diff
@@ -3,11 +3,25 @@ import '../domains/common/components/styles/global.scss'
...
 import { Provider } from 'react-redux'
+import React, { PropsWithChildren } from 'react'
+
+export interface ContainerPage<PropsType> extends React.FC {
+    container: React.FC<PropsType>
+}
+
+interface CustomAppProps extends AppProps {
+    Component: ContainerPage<PropsWithChildren<any>>,
+}
```
Comment (levil664) [inline apps/schools/pages/_app.tsx]: interfaces folder?

File: apps/schools/pages/auth/forgot.tsx [hunk 49]
```diff
@@ -0,0 +1,109 @@
...
+            onReset={() => {
+                setStep('inputPhone')
+                Router.push('/auth/forgot')
+            }}
+            title={'Восстановление пароля'}
+        />,
+        'reset': <>
+            <ResetForm onFinish={() => {}}/>
+        </>,
+    }
```
Comment (levil664) [inline apps/schools/pages/auth/forgot.tsx]: constants folder?

**Review by levil664 — COMMENTED** [review]
image
image

why, when restoring a password, I have to register, and not just restore the password according to the figma

**Review by levil664 — COMMENTED** [review]
image
image

why, when restoring a password, I have to register, and not just restore the password according to the figma

**Review by levil664 — CHANGES_REQUESTED** [review]
there are too many comments they should be filtered out or deleted

File: apps/schools/domains/user/components/containers/AuthLayout.tsx [hunk 50]
```diff
@@ -0,0 +1,73 @@
...
+        <GoogleReCaptchaProvider
+            reCaptchaKey={googleCaptcha && googleCaptcha.SITE_KEY}
+            useRecaptchaNet
+            container={GOOGLE_RECAPTCHA_CONTAINER}
+            scriptProps={GOOGLE_RECAPTCHA_SCRIPT_PROPS}>
+
+            <div id='ReCaptchaContainer'/>
+            <div
+                className={styles.container}
+                style={{ backgroundImage: 'url(/image/authImage.svg)' }}
```
Comment (levil664) [inline apps/schools/domains/user/components/containers/AuthLayout.tsx]: scss?

File: apps/schools/domains/common/components/Button/index.tsx (hunk 4 above)
Comment (mzavarka) [reply apps/schools/domains/common/components/Button/index.tsx]: we don't use them but
it is example styles

**Review by levil664 — APPROVED** [review]

**Review by levil664 — APPROVED** [review]

**Pull request merged**